packages/xstate-react/src/useMachine.ts

```diff
@@ -118,7 +119,7 @@ export function useMachine<TMachine extends AnyStateMachine>(
         ? ((service.behavior as AnyStateMachine).createState(
             rehydratedState
           ) as any)
-        : storeSnapshot
+        : undefined
```

Comment:
review note: `storeSnapshot` is what is internally returned by `this.getInitialState()` (as internally that initial state is cached). The difference is now in what `.start(arg)` might do based on the explicit argument - as the rehydrated state would go through `this.behavior.restoreState`. This in turn might create a copy of the state argument - maybe change its values somehow etc.